async_task.go

```diff
@@ -41,6 +42,15 @@ var ErrTimeout = errors.New("timeout")
 // ErrCanceled is returned if a cancel is triggered
 var ErrCanceled = errors.New("canceled")
 
+var CompletedTask = &TaskStatus{
```

Comment:
you probably want these private.
The consumer of the package should not be able to mutate the CompletedTask variable from outside your package?